language/tools/move-package/src/compilation/compiled_package.rs

```diff
@@ -342,7 +346,10 @@ impl OnDiskCompiledPackage {
             category_dir
                 .join(&file_path)
                 .with_extension(MOVE_COMPILED_EXTENSION),
-            compiled_unit.unit.serialize().as_slice(),
+            compiled_unit
+                .unit
+                .serialize(&get_bytecode_version_from_env())
```

Comment:
Same for the other package one, this feels like it should be in the Move.toml

A todo + issue might be good here

Reply:
+1

Reply:
See #129 . The current design of cli/package system makes it real hard to do this. It already bothered me before. here is a deeper refactoring needed.